File: BC-W4/ArmyJava/src/abilities/SoldierAbility.java
package abilities;

import units.Unit;
import exceptions.*;

public class SoldierAbility extends Ability {
    public SoldierAbility(Unit owner) {
        super(owner);
    }

    public void transform() throws InvalidTransformException {
        throw new InvalidTransformException();
    }

    public void infect(Unit enemy) throws InvalidInfectException {
        throw new InvalidInfectException();
    }
}

File: BC-W4/Army/Abilities/SoldierAbility/SoldierAbility.cpp
#include "SoldierAbility.h"

SoldierAbility::SoldierAbility(Unit* unit) : Ability(unit) {};
SoldierAbility::~SoldierAbility() {};

void SoldierAbility::castAbility() {
    throw HasNoSuchAbilityException();
};

void SoldierAbility::castAbility(Unit* enemy) {
    throw HasNoSuchAbilityException();
};